code/obj/playable_piano.dm

```diff
@@ -207,6 +212,15 @@ TYPEINFO(/obj/player_piano)
 					return 1
 		return 0
 
+	proc/is_pulser_auto_linking(var/mob/M)
+		if(hasvar(M, "l_hand") && ispulsingtool(M.l_hand) && SEND_SIGNAL(M.l_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(hasvar(M, "r_hand") && ispulsingtool(M.r_hand) && SEND_SIGNAL(M.r_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(hasvar(M, "module_states"))
```

Comment:
no, typecheck

Reply:
I added typechecks, though I am unsure if I understood correctly.

Reply:
no, as in don't use `hasvar`